### ISSUE TITLE
Quarantine E2E culture-aware tests

### ISSUE THREAD

**Issue by sebastienros** [dotnet/aspnetcore#41425]
Unblocking https://github.com/dotnet/aspnetcore/pull/41158

Issue has been identified in SDK/runtime trimming logic. Quarantining the failing tests since we don't want to block SDK ingestion for too long because of that.

## Build

https://dev.azure.com/dnceng/public/_build/results?buildId=1726505&view=results

**Comment by dougbu** [comment]
@sebastienros what's the latest status on the trimming issue. Is there for example a dotnet/runtime PR or issue we should track here❔

**Comment by halter73** [comment]
@javiercn @SteveSandersonMS I think investigating this is asking a bit much out of whoever happens to be on dev ops rotation since it involves flakiness browser testing. I think this benefits from some domain experties. I'm going to remove it from the rotation tracking on the next hand off.

https://dev.azure.com/dnceng/public/_test/analytics?definitionId=1026&contextType=build

The pass rate is currently well under 50% for a lot of the failing tests. It looks like they either pass or fail in bunches. 

![image](https://user-images.githubusercontent.com/54385/167231732-f1d8606e-36c6-451b-b229-e14de4cf8a4d.png)

**Comment by SteveSandersonMS** [comment]
> Issue has been identified in SDK/runtime trimming logic.

@sebastienros Could you point to whichever external issue or conversations identified this cause? @lewing Do you know if there's an ETA on a fix for this?

**Comment by sebastienros** [comment]
@SteveSandersonMS there is a chat room in Teams with all the details and links to investigations.
Pinging you in it for visibility.

**Comment by SteveSandersonMS** [comment]
@lewing Interesting. Do you have a non-wasm repro?

**Comment by lewing** [comment]
Let me rephrase that, I don't think this is related to any changes specific to browser-wasm in runtime, the failure might uniquely present there.

**Comment by lewing** [comment]
Is it failing on main the same way it was failing originally? Can someone on the aspnetcore team trace the assertion back to the first place you see an incorrect value?

**Comment by SteveSandersonMS** [comment]
Yes, it fails everywhere now we've taken the runtime/SDK update.

I investigated and it's very easy to repro independently of any Blazor code. It's simply that **all date parsing is broken after trimming** (and maybe other culture-specific operations).

Example:

```cs
var parsed = DateTime.Parse("6/27/2022 11:08:05 AM");
```

Expected: Well, you know
Actual:

```
dotnet.7.0.0-preview.5.22274.8.ekt40wn8gu.js:3 
        
       Error: [MONO] * Assertion: should not be reached at /__w/1/s/src/mono/mono/mini/interp/interp.c:3606

    at mono_wasm_stringify_as_error_with_stack (dotnet.7.0.0-preview.5.22274.8.ekt40wn8gu.js:3:19927)
    at Object.mono_wasm_trace_logger (dotnet.7.0.0-preview.5.22274.8.ekt40wn8gu.js:3:20298)
    at _mono_wasm_trace_logger (dotnet.7.0.0-preview.5.22274.8.ekt40wn8gu.js:12:105856)
    at dotnet.wasm:0x9eb2
    at dotnet.wasm:0xa21ed
    at dotnet.wasm:0x1bed4d
    at dotnet.wasm:0x1beddc
    at dotnet.wasm:0x1bee45
    at dotnet.wasm:0xd5fd
    at dotnet.wasm:0xcab0
```

I've filed https://github.com/dotnet/runtime/issues/69891

**Comment by lewing** [comment]
@SteveSandersonMS but only datetime is failing now?

**Comment by SteveSandersonMS** [comment]
@lewing I don't know how much functionality is broken in total. It's quite possible a lot of culture-sensitive logic is broken but that will most likely become clearer once the underlying fault and fix is identified. At that point we can see if all the E2E tests go green again.

**Comment by lewing** [comment]
This appears to have been a linker bug in the version specified by the sdk.  @radical added the details to https://github.com/dotnet/runtime/issues/69891

In the future I'd suggest updating the linker when you update runtime or at least checking against the version used in runtime when there are problems updating either the sdk or runtime.

**Comment by SteveSandersonMS** [comment]
> In the future I'd suggest updating the linker when you update runtime or at least checking against the version used in runtime when there are problems updating either the sdk or runtime.

That's interesting. I wasn't aware we had control over the linker version explicitly. The SDK and runtime updates flow into this repo automatically via "darc". I had thought the linker version just came from the SDK. Are you saying this dependency flow isn't right and we should be updating something for the linker separately?

Or does this point to some more basic flaw like the fact that SDK and runtime have independent versions (and flow independently) means they can never be guaranteed to be mutually compatible? Should we look at only consuming the runtime *via* the SDK?

**Comment by javiercn** [comment]
I have re-triggered the PR to unquarantine them.
https://github.com/dotnet/aspnetcore/pull/41494

**Comment by javiercn** [comment]
@dougbu these tests weren't flaky, there was a bug in the runtime that triggered the failure. They should have been marked Skip, not Quarantined. I expect them to go back to working reliably now that the fix is technically merged.

**Comment by dougbu** [comment]
Thanks @javiercn 👍 

I got confused way back at @halter73's
> The pass rate is currently well under 50% for a lot of the failing tests.
